jbpm-task-assigning/kie-server-services-task-assigning/kie-server-services-task-assigning-planning/src/main/java/org/kie/server/services/taskassigning/planning/SolutionProcessor.java

```diff
@@ -0,0 +1,249 @@
+        int index;
+        int publishedCount;
+        for (User user : solution.getUserList()) {
+            userPlanningItems = new ArrayList<>();
+            index = 0;
+            publishedCount = 0;
+            Task nextTask = user.getNextTask();
+
+            while (nextTask != null) {
+                if (IS_NOT_DUMMY.test(nextTask)) {
+                    //dummy tasks has nothing to with the jBPM runtime, don't process them
+                    planningItem = PlanningItem.builder()
+                            .containerId(nextTask.getContainerId())
+                            .taskId(nextTask.getId())
+                            .processInstanceId(nextTask.getProcessInstanceId())
+                            .planningTask(PlanningTask.builder()
+                                                  .taskId(nextTask.getId())
+                                                  .published(nextTask.isPinned())
+                                                  .assignedUser(user.getUser().getEntityId())
+                                                  .index(index++)
+                                                  .build())
+                            .build();
+
+                    userPlanningItems.add(planningItem);
+                    publishedCount += planningItem.getPlanningTask().isPublished() ? 1 : 0;
+                }
+                nextTask = nextTask.getNextTask();
+            }
+            if (!IS_PLANNING_USER.test(user.getEntityId())) {
+                userPlanningItemsIt = userPlanningItems.iterator();
+                while (userPlanningItemsIt.hasNext() && publishedCount < publishWindowSize) {
+                    planningItem = userPlanningItemsIt.next();
+                    if (!planningItem.getPlanningTask().isPublished()) {
+                        planningItem.getPlanningTask().setPublished(true);
+                        publishedCount++;
+                    }
+                }
+            }
+            planningItems.addAll(userPlanningItems);
+        }
+
+        final List<PlanningItem> publishedTasks = planningItems.stream().filter(item -> item.getPlanningTask().isPublished()).collect(Collectors.toList());
+
+        if (LOGGER.isTraceEnabled()) {
+            traceSolution(solution);
+            tracePublishedTasks(publishedTasks);
+        }
+
+        Result result;
+        try {
+            ExecutePlanningResult executeResult = delegate.executePlanning(publishedTasks, targetUserId);
+            result = new Result(executeResult);
+        } catch (Exception e) {
+            LOGGER.error("An error was produced during solution processing, planning execution failed.", e);
+            result = new Result(e);
+        }
+
+        LOGGER.debug("Solution processing finished: " + solution);
+        processing.set(false);
+        resultConsumer.accept(result);
+    }
+
+    private void traceSolution(TaskAssigningSolution solution) {
+        LOGGER.trace("\n");
```

Comment:
Every call to LOGGER.trace will print a new trace message - with a new timestamp. I suspect we should rather format the entire solution string first and then just print it once.